test/wallet-test.js

```diff
@@ -2130,13 +2131,52 @@ describe('Wallet', function() {
         await mineBlock();
 
         await forValue(node.chain, 'height', height + 1);
-        assert.equal(wdb.height, height);
+        assert.equal(wdb.height, height + 1);
 
         height += 1;
       }
 
       for (let i = 0; i < 10; i++)
         await raceForward();
     });
+
+    it('should emit details with correct confirmation', async () => {
+      const wclient = new WalletClient({port: ports.wallet});
+      await wclient.open();
+
+      const info = await wclient.createWallet('test');
+      const wallet = wclient.wallet('test', info.token);
+      await wallet.open();
+
+      const acct = await wallet.getAccount('default');
+      const waddr = acct.receiveAddress;
+
+      miner.addresses.length = 0;
+      miner.addAddress(waddr);
+
+      let txCount = 0;
+      let confirmedCount = 0;
+
+      wallet.on('tx', (details) => {
+        if (details.confirmations === 1)
+          txCount += 1;
+      });
+
+      wallet.on('confirmed', (details) => {
+        if (details.confirmations === 1)
```

Comment:
I think you can assert that `confirmations === 1` here, then increment the counter on every event. Would help to catch future regressions (like the one we're about to add that emits `'confirmed'` events on coinbase TXs!)

Reply:
Yeah... possible. Asserting the count is correct has the same effect.

Reply:
Added to catch the case even without asserting the count.